crates/rnote-ui/src/appwindow/actions.rs

```diff
@@ -406,7 +406,19 @@ impl RnAppWindow {
         // Trash Selection
         action_selection_trash.connect_activate(clone!(@weak self as appwindow => move |_, _| {
             let canvas = appwindow.active_tab_wrapper().canvas();
-            let widget_flags = canvas.engine_mut().trash_selection();
+            // check if we have a selector as a temporary tool and need to change the pen
+            let cancel_selection = canvas.engine_ref().cancel_selection_temporary_pen();
+            let widget_flags = if cancel_selection {
+                // trigger an event for a KeyboardPress::Delete
+                let (_, widget_flags) = canvas.engine_mut().handle_pen_event(
+                    rnote_compose::PenEvent::KeyPressed { keyboard_key: KeyboardKey::Delete, modifier_keys: vec![] },
+                    None,
+                    Instant::now()
+                );
+                widget_flags
+            } else {
+                canvas.engine_mut().trash_selection()
+            };
```

Comment:
Shouldn' this logic be in the engine in the `trash_selection()` function?

Reply:
yes, it's only called from here anyway. I changed it and it may also make it easier to merge as well